packages/shared/pkg/storage/storage_google.go

```diff
@@ -222,6 +221,26 @@ func (g *GCPBucketStorageObjectProvider) WriteFromFileSystem(path string) error
 	objectName := g.path
 	filePath := path
 
+	fileInfo, err := os.Stat(filePath)
```

Comment:
wouldn't it make sense to have this as a part of the multipart uploader? If we ever decide to use it elsewhere, it make sense to do it the same way, right?